### PR TITLE
Configure system to support client certificate authentication

### PR DESCRIPTION
**Pull request by janan07**

### CODE REVIEW

File: docs/extend/extend-apiml/authentication-for-apiml-services.md [hunk 1]
```diff
@@ -105,22 +105,29 @@ with each authentication provider.
...
+The Gateway resolves authetication by authenticating against an endpoint through the following process:
 
 * The client calls the service endpoint through API ML Gateway with the client certificate.
 * The client certificate and private key are checked as a valid TLS client certificate against the trusted CAs of the Gateway.
-* The public part of the provided client certificate is checked against SAF. SAF subsequently returns a user ID that owns this certificate. ZSS provides this API for API ML. 
+* The public part of the provided client certificate is checked against SAF. SAF subsequently returns a user ID that owns this certificate. ZSS provides this API for API ML.
 * The Gateway performs the login of the mapped user and provides valid authentication to the southbound service.
 
-Authentication via the login endpoint is performed in the following way:
+The Gateway resolves authentication via the login endpoint through the following process:
```
Comment (pablocarle) [inline docs/extend/extend-apiml/authentication-for-apiml-services.md:130]: Similar to the previous one
```suggestion
When sending a request to the login endpoint with a client certificate, the Gateway performs the following process to exchange the client certificate for an authentication token.
```

File: docs/extend/extend-apiml/authentication-for-apiml-services.md [hunk 2]
```diff
@@ -129,40 +136,48 @@ Authentication via the login endpoint is performed in the following way:
 
 ![Zowe client certificate authentication diagram](../../images/api-mediation/zowe-client-cert-auth.png)
 
-### Procedure to authenticate with client certificates
-
-Follow these steps to authenticate with client certificates:
+#### Prerequisites
```
Comment (pablocarle) [inline docs/extend/extend-apiml/authentication-for-apiml-services.md:139]: Maybe there should be an introductory line:

"Make sure the following requirements are met before setting up client certificate authentication"

File: docs/extend/extend-apiml/authentication-for-apiml-services.md [hunk 3]
```diff
@@ -129,40 +136,48 @@ Authentication via the login endpoint is performed in the following way:
...
 
 1. Specify the Zowe runtime user and set your protection by password. The user is created with the `NOPASSWORD` parameter by the Zowe installer. It is necessary to change this password. For RACF, issue the following TSO command:  
 `ALTUSER <ZOWE_RUNTIME_USER (ZWESVUSR by default)> PASSWORD(<NEWPASSWORD>)`  
 For other security systems, refer to the documentation for an equivalent command.
 
+<!-- Add sample command and output -->
```
Comment (pablocarle) [inline docs/extend/extend-apiml/authentication-for-apiml-services.md:145]: The sample command for RACF is above

File: docs/extend/extend-apiml/authentication-for-apiml-services.md [hunk 4]
```diff
@@ -129,40 +136,48 @@ Authentication via the login endpoint is performed in the following way:
...
+
 2. Verify that the Zowe runtime user is allowed to log in to z/OSMF. (Check that the user is member of the default `IZUUSER` group.)
 
+<!-- Add how to provide this verification.. Sample commands? -->
```
Comment (pablocarle) [inline docs/extend/extend-apiml/authentication-for-apiml-services.md:149]: I guess this is a simple list user command

File: docs/extend/extend-apiml/authentication-for-apiml-services.md [hunk 5]
```diff
@@ -129,40 +136,48 @@ Authentication via the login endpoint is performed in the following way:
...
+To validate that the client certificate functionality works properly, call the login endpoint with the certificate that was setup using the steps in _Configure your z/OS system to support client certificate authentication_. 
+
+Validate using CURL, a command line utility that runs on Linux based systems:
 
 **Example:**
 ```
 curl --cert /path/to/cert.pem --key /path/to/key.pem https://api-mediation-layer:7554/gateway/api/v1/login
 ```
+<!-- Add outcome -->
+
```
Comment (pablocarle) [inline docs/extend/extend-apiml/authentication-for-apiml-services.md:180]: maybe "your Zowe instance is configured to accept x.509 client certificates authentication"

File: docs/extend/extend-apiml/authentication-for-apiml-services.md [hunk 6]
```diff
@@ -129,40 +136,48 @@ Authentication via the login endpoint is performed in the following way:
...
 #### Validate the client certificate functionality
 
-To validate that the client certificate functionality works properly, call the login endpoint with the certificate that was setup using the steps in the previous section, _Procedure to authenticate with client certificates_. The following example uses curl, which is generally available on Linux based systems. 
+To validate that the client certificate functionality works properly, call the login endpoint with the certificate that was setup using the steps in _Configure your z/OS system to support client certificate authentication_. 
+
+Validate using CURL, a command line utility that runs on Linux based systems:
 
 **Example:**
 ```
 curl --cert /path/to/cert.pem --key /path/to/key.pem https://api-mediation-layer:7554/gateway/api/v1/login
```
Comment (pablocarle) [inline docs/extend/extend-apiml/authentication-for-apiml-services.md:177]: This is the sample command I was talking about, it's already here as a means of verification of the setup.

File: docs/extend/extend-apiml/authentication-for-apiml-services.md (hunk 4 above)
Comment (janan07) [reply docs/extend/extend-apiml/authentication-for-apiml-services.md:149]: I'm not sure I understand. Is this is the same as validating the client certificate functionality?

Comment (pablocarle) [reply docs/extend/extend-apiml/authentication-for-apiml-services.md:149]: No I think this is talking about the previous statement, to verify that the server user (usually zwesvusr) has access to z/OSMF by being a member of the default `IZUUSER` group

**Review by pablocarle — APPROVED** [review]

File: docs/extend/extend-apiml/authentication-for-apiml-services.md (hunk 4 above)
Comment (janan07) [reply docs/extend/extend-apiml/authentication-for-apiml-services.md]: Should I state it like this? "For RACF, issue the following TSO command to verify that zwesvusr has access to z/OSMF."?
Can you please clarify?

**Pull request merged**